relayer/workers/parachaincommitmentrelayer/beefy-listener.go

```diff
@@ -75,81 +93,122 @@ func (li *BeefyListener) onDone(ctx context.Context) error {
 }
 
 func (li *BeefyListener) subBeefyJustifications(ctx context.Context) error {
-	ch := make(chan interface{})
+	headers := make(chan *gethTypes.Header, 5)
 
-	li.log.Info("Subscribing to relay chain light client for new mmr payloads")
-	sub, err := li.relaychainConn.GetAPI().Client.Subscribe(context.Background(), "beefy", "subscribeJustifications", "unsubscribeJustifications", "justifications", ch)
-	if err != nil {
-		panic(err)
-	}
-	defer sub.Unsubscribe()
+	li.ethereumConn.GetClient().SubscribeNewHead(ctx, headers)
```

Comment:
Do subscriptions from geth guarantee notifications? 

If they don't we we might miss beefy justifications, which could halt the relayer. Generally, polling with timeouts is a bit more robust than subscriptions, unless the subscription itself has fallback/recovery polling implemented.

Reply:
I think they are supposed to guarantee notifications yes, though in practice I'm sure they are not that unreliable.

Graceful recovery on crash/restart and possibly also polling will be implemented in the catchup PR, so will leave this for then